### ISSUE TITLE
Run a community Thanos demo instance

### ISSUE THREAD

**Issue by onprem** [thanos-io/thanos#4606]
Prometheus runs a publically accessible instance, which they are using for CI to test the React UI for every pull request.

Running a public instance of Thanos will allow us to:
- demonstrating functionality very quickly to new users
- be more helpful in community support, where we can reference the demo instance UI
- can act as a playground for PromQL and Thanos component metrics
- can be used in the CI for testing the UI related changes (this means that contributors won't have to post any screenshots)

cc @thanos-io/thanos-maintainers

**Comment by yeya24** [comment]
I was thinking about the same thing before. One demo site for Prometheus is sufficient as it is just a single application. For thanos, do we want to have multiple demo sites for different components, querier, bucket UI, etc?

**Comment by onprem** [comment]
> Do you mean this? https://github.com/prometheus/demo-site
> 
> I'm curious how the Prometheus folks experience these demo sites (Like does it get used a lot for example?)

Yes, that's what I meant.

> For thanos, do we want to have multiple demo sites for different components, querier, bucket UI, etc?

Yep, we need to have multiple components running for having a proper demo of Thanos.

**Comment by fktkrt** [comment]
Great idea, I'd be happy to help!
Isn't an exposed Thanos Query enough at first? You could run queries and check the stores/config/etc.

Bucketweb could also make sense, but as a separate app (e.g. as Alertmanager on the aforementioned Prometheus demo site)? It's not that interactive/interesting in my opinion.

**Comment by GiedriusS** [comment]
I think for this we'd need query-frontend + Query + Store + Sidecar + Prometheus with a reverse proxy in front of query-frontend. Feel free to work on Ansible playbooks, later on we could move them under the `thanos-io` community & work on adding automated deployment.

**Comment by 007vedant** [comment]
Hey @GiedriusS ! I'd love to work on this issue under LFX Mentorship'22.

**Comment by ggold7046** [comment]
Hi, I am new to open source project.  Could anyone guide me how to contribute to this project. I have some knowledge of git,
linux , ansible

**Comment by GiedriusS** [comment]
Please apply via the LFX website! Thanks.

**Comment by wiardvanrij** [comment]
@tend2infinity

**Comment by tend2infinity** [comment]
Hey, I'll be working on it!

**Comment by halcyondude** [comment]
There's an opportunity to collaborate with a workstream launched out of https://github.com/cncf/tag-observability [WG: observe-k8s](https://github.com/cncf/tag-observability/projects/2)

https://github.com/observe-k8s/hub/blob/main/charter.md#vision

How/where can I learn more about this effort?

Thanks!
Matt

**Comment by fktkrt** [comment]
Are the infra and tooling decided yet? There are mentions of Ansible, but IIRC observe-k8s is aiming to use tools that fit k8s use cases better.

**Comment by wiardvanrij** [comment]
> There's an opportunity to collaborate with a workstream launched out of https://github.com/cncf/tag-observability ([WG: observe-k8s](https://github.com/cncf/tag-observability/projects/2))
> 
> https://github.com/observe-k8s/hub/blob/main/charter.md#vision
> 
> How/where can I learn more about this effort?
> 
> Thanks! Matt

Hi, we've got this LFX project and @GiedriusS and me are mentors for @tend2infinity who is mainly working on this. 

So, there are a few high level things that we have as information. Is there anything specific? If you want we can organize a small meeting to align things.

**Comment by wiardvanrij** [comment]
> Are the infra and tooling decided yet? There are mentions of Ansible, but IIRC observe-k8s is aiming to use tools that fit k8s use cases better.

Yes and no. The Ansible stuff and everything was based on what resources we had available. However, I felt the same and reached out to Civo. We are now getting sponsored for a kubernetes cluster :)

**Comment by tend2infinity** [comment]
Still relevant
